src/nl/hannahsten/texifyidea/util/labels/LabelExtraction.kt

```diff
@@ -65,44 +64,54 @@ fun PsiElement.extractLabelElement(): PsiElement? {
+ *
+ * Use `PsiFile.findCommandInFileSet(LatexGenericRegularCommand.EXTERNALDOCUMENT)` to get the optional parameter
+ *
+ *
+ * @param externalDocumentCommand the command that is used to include labels from other filesets, such as `\externaldocument`.
+ *
+ * @param referencingFileSetCommands a map of `LatexCommand.name` to `LatexCommand`.
+ * All the commands in the fileset containing the element referencing [this], which can make a different for what the label name should be, as using the xr package labels from other filesets can be included with a prefix.
+ *
+ *
  */
-fun PsiElement.extractLabelName(referencingFileSetCommands: Collection<LatexCommands>? = null): String {
-    return when (this) {
-        is BibtexEntry -> this.getIdentifier()
+fun PsiElement.extractLabelName(externalDocumentCommand: LatexCommands? = null): String {
+    when (this) {
+        is BibtexEntry -> return this.getIdentifier()
         is LatexCommands -> {
             if (CommandMagic.labelAsParameter.contains(name)) {
-                getOptionalParameterMap().toStringMap()["label"] ?: ""
+                return getOptionalParameterMap().toStringMap()["label"] ?: ""
             }
-            else {
-                // For now just take the first label name (which may be multiple for user defined commands)
-                val info = CommandManager.labelAliasesInfo.getOrDefault(name, null)
-                val position = info?.positions?.firstOrNull() ?: 0
-                var prefix = info?.prefix ?: ""
+            // For now just take the first label name (which may be multiple for user defined commands)
+            val info = CommandManager.labelAliasesInfo.getOrDefault(name, null)
+            val position = info?.positions?.firstOrNull() ?: 0
+            var prefix = info?.prefix ?: ""
 
-                // Check if there is any prefix given by the xr package
-                if (referencingFileSetCommands != null) {
-                    referencingFileSetCommands
-                        // Don't think there can be multiple, at least I wouldn't know what prefix it would use
-                        .firstOrNull { it.name == LatexGenericRegularCommand.EXTERNALDOCUMENT.commandWithSlash }
-                        ?.parameterList?.firstNotNullOfOrNull { it.optionalParam }
-                        ?.text?.trim('[', ']')
-                        ?.let { prefix = it }
-                }
-
-                // Skip optional parameters for now (also below and in
-                prefix + this.requiredParameter(position)
-            }
+            // Check if there is any prefix given by the xr package
+            externalDocumentCommand?.parameterList?.firstNotNullOfOrNull { it.optionalParam }
+                ?.text?.trim('[', ']')
+                ?.let { prefix = it }
+            // The original code here:
```

Comment:
I can see the original code in the git diff, no need to leave it here. In fact, that makes reviewing the diff more difficult